packages/blocks/github-pr-overview/src/app.tsx

```diff
@@ -1,241 +1,296 @@
+            payload: {
+              allPrs: prs,
+              githubEntityTypeIds: entityTypeIds,
+              blockState: BlockState.Selector,
+            },
+          });
+        })
+        .catch((err) => {
+          dispatch({
+            type: "UPDATE_STATE",
+            payload: {
+              blockState: BlockState.Error,
+              infoMessage: err.message,
+            },
+          });
+        });
     }
-  }, [
-    githubEntityTypeIds,
-    selectedPullRequestId,
-    aggregateEntities,
-    accountId,
-    setPullRequest,
-    setBlockState,
-  ]);
+  }, [graphService, githubEntityTypeIds, allPrs]);
 
+  // Fetch PR Details => pullRequest, events and reviews
+  // if there's a selectedPullRequestId
   React.useEffect(() => {
+    if (!blockRef.current) return;
+    // @todo add check to see if selectedPR is the same as current PR
     if (
-      selectedPullRequestId !== undefined &&
-      githubEntityTypeIds !== undefined
+      selectedPullRequestId &&
+      githubEntityTypeIds &&
+      graphService?.aggregateEntities
     ) {
-      setBlockState(BlockState.Loading);
-      collectReviewsAndSetState(
-        githubEntityTypeIds[GITHUB_ENTITY_TYPES.Review],
-        aggregateEntities,
-        accountId,
-        1,
-        setReviews,
+      dispatch({
+        type: "UPDATE_STATE",
+        payload: {
+          blockState: BlockState.Loading,
+          infoMessage: `Creating your timeline for pull request #${selectedPullRequestId.number}`,
+        },
+      });
+
+      void getPRDetails(
         selectedPullRequestId,
-      );
+        githubEntityTypeIds,
+        ({ data }) => graphService?.aggregateEntities({ data }),
+      )
+        .then((data) => {
+          dispatch({
+            type: "UPDATE_STATE",
+            payload: {
+              pullRequest: data.pullRequest,
+              reviews: data.reviews,
+              events: data.events,
+              blockState: BlockState.Overview,
+            },
+          });
+        })
+        .catch((err) => {
+          dispatch({
+            type: "UPDATE_STATE",
+            payload: {
+              blockState: BlockState.Error,
+              infoMessage: err.message,
+            },
+          });
+        });
     }
-  }, [
-    githubEntityTypeIds,
-    selectedPullRequestId,
-    aggregateEntities,
-    accountId,
-    setReviews,
-    setBlockState,
-  ]);
-  React.useEffect(() => {
-    if (
-      selectedPullRequestId !== undefined &&
-      githubEntityTypeIds !== undefined
-    ) {
-      setBlockState(BlockState.Loading);
-      collectPrEventsAndSetState(
-        githubEntityTypeIds[GITHUB_ENTITY_TYPES.IssueEvent],
-        aggregateEntities,
-        accountId,
-        1,
-        setEvents,
-        selectedPullRequestId,
-      );
+  }, [githubEntityTypeIds, selectedPullRequestId, graphService]);
+
+  /** @todo - Figure out when to query for more than one page, probably querying until no more results */
```

Comment:
Is this todo still necessary? It seems weirdly placed

Reply:
It no longer is. Thanks for catching